lib/abilities.ex

```diff
@@ -9,5 +9,25 @@ defimpl Canada.Can, for: Pairmotron.User do
   def can?(%User{id: user_id}, action, %Group{owner_id: user_id})
     when action in [:edit, :update, :delete], do: true
 
+  def can?(%User{id: user_id}, :show, project = %Project{}) do
+    if is_nil(project.group_id) do
+      false
+    else
+      project = project |> Pairmotron.Repo.preload([{:group, :users}])
+      project.group.users
+      |> Enum.any?(fn user -> user.id == user_id end)
+    end
+  end
```

Comment:
Is it possible to split this into two functions?

`def can?(%User{id: user_id}, :show, project = %Project{group_id: group_id}) when is_nil(group_id) do`?